app/src/androidTest/java/com/github/polypoly/app/map/VisitMapActivityTest.kt

```diff
@@ -78,13 +77,15 @@ class VisitMapActivityTest {
 
     @Test
     fun clickOnCloseButtonCloseTheDialog() {
-        val marker = MapUI.mapView.overlays.first { it is Marker} as Marker
-        composeTestRule.activity.mapViewModel.selectedMarker = marker
-        composeTestRule.activity.interactingWithProperty.value = true
+        setLocationInMapViewModel(getRandomLocation()).get(TIMEOUT_DURATION, TimeUnit.SECONDS)
         runBlocking { delay(500) }
```

Comment:
for all these `runBlocking { delay(500) }` you can use the function `waitForUIToUpdate` that @AmLibra do in `GameActivityTest` and put this function in a file in the package `commons`

Reply:
Yep it is definitely clearer to do that !